pkg/scalers/mssql_scaler.go

```diff
@@ -0,0 +1,280 @@
+		meta.connectionString = config.AuthParams["connectionString"]
+	case config.TriggerMetadata["connectionStringFromEnv"] != "":
+		meta.connectionString = config.ResolvedEnv[config.TriggerMetadata["connectionStringFromEnv"]]
+	default:
+		meta.connectionString = ""
+		if val, ok := config.TriggerMetadata["host"]; ok {
+			meta.host = val
+		} else {
+			return nil, fmt.Errorf("no host given")
+		}
+
+		if val, ok := config.TriggerMetadata["port"]; ok {
+			port, err := strconv.Atoi(val)
+			if err != nil {
+				return nil, fmt.Errorf("port parsing error %s", err.Error())
+			}
+
+			meta.port = port
+		}
+
+		if val, ok := config.TriggerMetadata["username"]; ok {
+			meta.username = val
+		}
+
+		// database is optional in SQL s
+		if val, ok := config.TriggerMetadata["database"]; ok {
+			meta.database = val
+		}
+
+		if config.AuthParams["password"] != "" {
+			meta.password = config.AuthParams["password"]
+		} else if config.TriggerMetadata["passwordFromEnv"] != "" {
+			meta.password = config.ResolvedEnv[config.TriggerMetadata["passwordFromEnv"]]
+		}
+	}
+
+	// get the metricName, which can be explicit or from the (masked) connection string
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
+		meta.metricName = kedautil.NormalizeString(fmt.Sprintf("mssql-%s", val))
+	} else {
+		if meta.database != "" {
+			meta.metricName = kedautil.NormalizeString(fmt.Sprintf("mssql-%s", meta.database))
+		} else if meta.host != "" {
+			meta.metricName = kedautil.NormalizeString(fmt.Sprintf("mssql-%s", meta.host))
+		} else if meta.connectionString != "" {
+			// The mssql provider supports of a variety of connection string formats. Instead of trying to parse
+			// the connection string and mask out sensitive data, play it safe and just hash the whole thing.
+			connectionStringHash := sha256.Sum256([]byte(meta.connectionString))
+			meta.metricName = kedautil.NormalizeString(fmt.Sprintf("mssql-%x", connectionStringHash))
+		} else {
+			meta.metricName = "mssql"
+		}
+	}
+
+	return &meta, nil
+}
+
+// newMSSQLConnection returns a new, opened SQL connection for the provided mssqlMetadata
+func newMSSQLConnection(meta *mssqlMetadata) (*sql.DB, error) {
+	connStr := getMSSQLConnectionString(meta)
+
+	db, err := sql.Open("sqlserver", connStr)
+	if err != nil {
+		mssqlLog.Error(err, fmt.Sprintf("Found error opening mssql: %s", err))
+		return nil, err
+	}
+
+	err = db.Ping()
+	if err != nil {
+		mssqlLog.Error(err, fmt.Sprintf("Found error pinging mssql: %s", err))
+		return nil, err
+	}
+
+	return db, nil
+}
+
+// getMSSQLConnectionString returns a connection string from a mssqlMetadata
+func getMSSQLConnectionString(meta *mssqlMetadata) string {
+	var connStr string
+
+	if meta.connectionString != "" {
+		connStr = meta.connectionString
+	} else {
+		query := url.Values{}
+		if meta.database != "" {
+			query.Add("database", meta.database)
+		}
+
+		connectionUrl := &url.URL{Scheme: "sqlserver", RawQuery: query.Encode()}
```

Comment:
nit: from the static checks, this should be `connectionURL`

Reply:
Those darn C# habits... ;)